clusterman/autoscaler/pool_manager.py

```diff
@@ -192,26 +192,34 @@ def prune_excess_fulfilled_capacity(
 
         if not dry_run:
             if self.draining_enabled:
-                assert self.draining_client  # make mypy happy
-                for group_id, node_metadatas in marked_nodes_by_group.items():
+                for node_metadatas in marked_nodes_by_group.values():
                     for node_metadata in node_metadatas:
-                        self.draining_client.submit_instance_for_draining(
-                            node_metadata.instance,
-                            sender=cast(
-                                Type[AWSResourceGroup],
-                                self.resource_groups[group_id].__class__,
-                            ),
-                            scheduler=self.scheduler,
-                            pool=self.pool,
-                            agent_id=node_metadata.agent.agent_id,
-                            draining_start_time=arrow.now(),
-                        )
+                        self.submit_for_draining(node_metadata)
             else:
                 for group_id, node_metadatas in marked_nodes_by_group.items():
                     self.resource_groups[group_id].terminate_instances_by_id(
                         [node_metadata.instance.instance_id for node_metadata in node_metadatas]
                     )
 
+    def submit_for_draining(self, node_metadata: ClusterNodeMetadata) -> None:
+        """Submit collection of nodes for draining
+
+        :param ClusterNodeMetadata node_metadata: node to be drained
+        """
+        assert self.draining_client  # make mypy happy
+        group_id = node_metadata.instance.group_id
```

Comment:
As far as I know, this should always be set, so getting it from `marked_nodes_by_group` was superfluous. Otherwise I can just make it an optional parameter to the method to maintain as much of the existing logic as possible.

Reply:
Yeah, I think so. Let's keep eyes on logs in deployment time, just in case.